### PR TITLE
Fixing Container bleed and alignment issues

### PR DESCRIPTION
**Pull request by pavankumarj**
# Description
Fixed the following scenarios
- TextBlock margin issue if placed directly inside AdaptiveCard
- Conatiner Bleed not working as expected
- Image alignment issues when image size is more than layout size
- Parent styles not derived when default style is used

# Sample Card

Sample JSON 1
`{
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "type": "AdaptiveCard",
    "version": "1.0",
    "body": [
        {
            "type": "Container",
            "style": "emphasis",
            "bleed": true,
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Bleeding container at the top of the card",
                    "wrap": true
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "Dummy TextBlock"
        },
        {
            "type": "Container",
            "style": "emphasis",
            "bleed": true,
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Bleeding container in the middle of the card",
                    "wrap": true
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "Dummy TextBlock"
        },
        {
            "type": "Container",
            "style": "emphasis",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Non-bleeding container",
                    "wrap": true
                },
                {
                    "type": "Container",
                    "style": "default",
                    "bleed": true,
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "Bleeding sub-container with style explicitly set to default",
                            "wrap": true
                        }
                    ]
                },
                {
                    "type": "Container",
                    "style": "default",
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "Non-bleeding sub-container with style explicitly set to default",
                            "wrap": true
                        }
                    ]
                },
                {
                    "type": "Container",
                    "bleed": true,
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "Bleeding sub-container with no explicit style",
                            "wrap": true
                        }
                    ]
                },
                {
                    "type": "Container",
                    "style": "good",
                    "bleed": true,
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "Bleeding sub-container with no explicit style",
                            "wrap": true
                        }
                    ]
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "Bleeding ColumnSet below"
        },
        {
            "type": "ColumnSet",
            "style": "emphasis",
            "bleed": true,
            "columns": [
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "Image",
                            "url": "https://pbs.twimg.com/profile_images/3647943215/d7f12830b3c17a5a9e4afcc370e3a37e_400x400.jpeg",
                            "size": "small",
                            "style": "person"
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "verticalContentAlignment": "center",
                    "items": [
                        {
                            "type": "TextBlock",
                            "horizontalAlignment": "center",
                            "text": "Middle column",
                            "wrap": true
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "verticalContentAlignment": "center",
                    "items": [
                        {
                            "type": "TextBlock",
                            "horizontalAlignment": "right",
                            "text": "Right column",
                            "wrap": true
                        }
                    ]
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "Left and right column bleeding only"
        },
        {
            "type": "ColumnSet",
            "columns": [
                {
                    "type": "Column",
                    "width": "stretch",
                    "style": "emphasis",
                    "bleed": true,
                    "items": [
                        {
                            "type": "Image",
                            "url": "https://pbs.twimg.com/profile_images/3647943215/d7f12830b3c17a5a9e4afcc370e3a37e_400x400.jpeg",
                            "size": "small",
                            "style": "person"
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "verticalContentAlignment": "center",
                    "items": [
                        {
                            "type": "TextBlock",
                            "horizontalAlignment": "center",
                            "text": "Middle column",
                            "wrap": true
                        },
                        {
                            "type": "Container",
                            "style": "emphasis",
                            "spacing": "none",
                            "bleed": true,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Bleeding sub-container in center column",
                                    "wrap": true,
                                    "horizontalAlignment": "center"
                                }
                            ]
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "style": "emphasis",
                    "bleed": true,
                    "verticalContentAlignment": "center",
                    "items": [
                        {
                            "type": "TextBlock",
                            "horizontalAlignment": "right",
                            "text": "Right column",
                            "wrap": true
                        }
                    ]
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "Dummy TextBlock"
        },
        {
            "type": "Container",
            "style": "emphasis",
            "bleed": true,
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Bleeding container",
                    "wrap": true
                },
                {
                    "type": "Container",
                    "style": "default",
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "Non-bleeding sub-container",
                            "wrap": true
                        },
                        {
                            "type": "Container",
                            "style": "emphasis",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Non-bleeding sub-sub-container",
                                    "wrap": true
                                },
                                {
                                    "type": "Container",
                                    "style": "default",
                                    "items": [
                                        {
                                            "type": "TextBlock",
                                            "text": "Non-bleeding sub-sub-sub-container containing bleeding ColumnSet",
                                            "wrap": true
                                        },
                                        {
                                            "type": "ColumnSet",
                                            "style": "emphasis",
                                            "bleed": true,
                                            "columns": [
                                                {
                                                    "type": "Column",
                                                    "width": "stretch",
                                                    "items": [
                                                        {
                                                            "type": "Image",
                                                            "url": "https://pbs.twimg.com/profile_images/3647943215/d7f12830b3c17a5a9e4afcc370e3a37e_400x400.jpeg",
                                                            "size": "small",
                                                            "style": "person"
                                                        }
                                                    ]
                                                },
                                                {
                                                    "type": "Column",
                                                    "width": "stretch",
                                                    "verticalContentAlignment": "center",
                                                    "items": [
                                                        {
                                                            "type": "TextBlock",
                                                            "horizontalAlignment": "center",
                                                            "text": "Middle column",
                                                            "wrap": true
                                                        }
                                                    ]
                                                },
                                                {
                                                    "type": "Column",
                                                    "width": "stretch",
                                                    "verticalContentAlignment": "center",
                                                    "items": [
                                                        {
                                                            "type": "TextBlock",
                                                            "horizontalAlignment": "right",
                                                            "text": "Right column",
                                                            "wrap": true
                                                        }
                                                    ]
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Non-bleeding container",
                    "wrap": true
                },
                {
                    "type": "Container",
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "Non-bleeding sub-container",
                            "wrap": true
                        },
                        {
                            "type": "Container",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Non-bleeding sub-sub-container",
                                    "wrap": true
                                },
                                {
                                    "type": "Container",
                                    "items": [
                                        {
                                            "type": "TextBlock",
                                            "text": "Non-bleeding sub-sub-sub-container containing bleeding ColumnSet",
                                            "wrap": true
                                        },
                                        {
                                            "type": "ColumnSet",
                                            "style": "emphasis",
                                            "bleed": true,
                                            "columns": [
                                                {
                                                    "type": "Column",
                                                    "width": "stretch",
                                                    "items": [
                                                        {
                                                            "type": "Image",
                                                            "url": "https://pbs.twimg.com/profile_images/3647943215/d7f12830b3c17a5a9e4afcc370e3a37e_400x400.jpeg",
                                                            "size": "small",
                                                            "style": "person"
                                                        }
                                                    ]
                                                },
                                                {
                                                    "type": "Column",
                                                    "width": "stretch",
                                                    "verticalContentAlignment": "center",
                                                    "items": [
                                                        {
                                                            "type": "TextBlock",
                                                            "horizontalAlignment": "center",
                                                            "text": "Middle column",
                                                            "wrap": true
                                                        }
                                                    ]
                                                },
                                                {
                                                    "type": "Column",
                                                    "width": "stretch",
                                                    "verticalContentAlignment": "center",
                                                    "items": [
                                                        {
                                                            "type": "TextBlock",
                                                            "horizontalAlignment": "right",
                                                            "text": "Right column",
                                                            "wrap": true
                                                        }
                                                    ]
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "Dummy TextBlock"
        },
        {
            "type": "Container",
            "style": "emphasis",
            "bleed": true,
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Bleeding container at the bottom of the card",
                    "wrap": true
                }
            ]
        }
    ]
}`

Sample JSON 2
`{
    "type": "AdaptiveCard",
    "body": [
        {
            "type": "Container",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Container - Style Not Set",
                    "wrap": true,
                    "weight": "Default",
                    "size": "Medium",
                    "color": "Accent"
                },
                {
                    "type": "ColumnSet",
                    "separator": true,
                    "columns": [
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Container",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cdn.sanity.io/images/0e4kwcjv/production/df506530768d939ccccb680d4f6f146aca0857d4-1664x1057.png?fm=webp"
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Text in nested container (Container-ColumnSet-Column-Container). Style NotSet",
                                            "wrap": true,
                                            "weight": "Bolder",
                                            "size": "Medium",
                                            "spacing": "Medium"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://cdn.sanity.io/images/0e4kwcjv/production/0f0a18e6dcf652c09dfd68d5d613499758b9b5a2-1664x1057.png?fm=webp"
                                },
                                {
                                    "type": "TextBlock",
                                    "text": "Another column (Container-ColumnSet-Column). Style NotSet",
                                    "wrap": true,
                                    "size": "Medium",
                                    "weight": "Bolder",
                                    "spacing": "Medium"
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "style": "default",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Container - Style Default",
                    "wrap": true,
                    "weight": "Default",
                    "size": "Medium",
                    "color": "Accent"
                },
                {
                    "type": "ColumnSet",
                    "separator": true,
                    "style": "default",
                    "columns": [
                        {
                            "type": "Column",
                            "width": "stretch",
                            "style": "default",
                            "items": [
                                {
                                    "type": "Container",
                                    "style": "default",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cdn.sanity.io/images/0e4kwcjv/production/df506530768d939ccccb680d4f6f146aca0857d4-1664x1057.png?fm=webp"
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Text in nested container (Container-ColumnSet-Column-Container). Style Default",
                                            "wrap": true,
                                            "weight": "Bolder",
                                            "size": "Medium",
                                            "spacing": "Medium"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "style": "default",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://cdn.sanity.io/images/0e4kwcjv/production/0f0a18e6dcf652c09dfd68d5d613499758b9b5a2-1664x1057.png?fm=webp"
                                },
                                {
                                    "type": "TextBlock",
                                    "text": "Another column (Container-ColumnSet-Column). Style Default",
                                    "wrap": true,
                                    "size": "Medium",
                                    "weight": "Bolder",
                                    "spacing": "Medium"
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "style": "default",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Container - Style Default",
                    "wrap": true,
                    "weight": "Default",
                    "size": "Medium",
                    "color": "Accent"
                },
                {
                    "type": "ColumnSet",
                    "separator": true,
                    "style": "default",
                    "columns": [
                        {
                            "type": "Column",
                            "width": "stretch",
                            "style": "accent",
                            "items": [
                                {
                                    "type": "Container",
                                    "style": "default",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cdn.sanity.io/images/0e4kwcjv/production/df506530768d939ccccb680d4f6f146aca0857d4-1664x1057.png?fm=webp"
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Text in nested container (Container(Default) - ColumnSet(Default) - Column(Accent) - Container(Default)",
                                            "wrap": true,
                                            "weight": "Bolder",
                                            "size": "Medium",
                                            "spacing": "Medium"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "style": "default",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://cdn.sanity.io/images/0e4kwcjv/production/0f0a18e6dcf652c09dfd68d5d613499758b9b5a2-1664x1057.png?fm=webp"
                                },
                                {
                                    "type": "TextBlock",
                                    "text": "Another column (Container-ColumnSet-Column). Style Default",
                                    "wrap": true,
                                    "size": "Medium",
                                    "weight": "Bolder",
                                    "spacing": "Medium"
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "style": "accent",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Container - Style Accent",
                    "wrap": true,
                    "weight": "Default",
                    "size": "Medium",
                    "color": "Accent"
                },
                {
                    "type": "ColumnSet",
                    "separator": true,
                    "style": "default",
                    "columns": [
                        {
                            "type": "Column",
                            "width": "stretch",
                            "style": "default",
                            "items": [
                                {
                                    "type": "Container",
                                    "style": "default",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cdn.sanity.io/images/0e4kwcjv/production/df506530768d939ccccb680d4f6f146aca0857d4-1664x1057.png?fm=webp"
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Text in nested container Container(Accent)-(ColumnSet-Column-Container)-Style Default",
                                            "wrap": true,
                                            "weight": "Bolder",
                                            "size": "Medium",
                                            "spacing": "Medium"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "style": "default",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://cdn.sanity.io/images/0e4kwcjv/production/0f0a18e6dcf652c09dfd68d5d613499758b9b5a2-1664x1057.png?fm=webp"
                                },
                                {
                                    "type": "TextBlock",
                                    "text": "Another column (Container(Accent)-ColumnSet-Column). Style Default",
                                    "wrap": true,
                                    "size": "Medium",
                                    "weight": "Bolder",
                                    "spacing": "Medium"
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "style": "Good",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Container - Style Good",
                    "wrap": true,
                    "weight": "Default",
                    "size": "Medium",
                    "color": "Good"
                },
                {
                    "type": "ColumnSet",
                    "separator": true,
                    "style": "default",
                    "columns": [
                        {
                            "type": "Column",
                            "width": "stretch",
                            "style": "attention",
                            "items": [
                                {
                                    "type": "Container",
                                    "style": "accent",
                                    "bleed": true,
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cdn.sanity.io/images/0e4kwcjv/production/df506530768d939ccccb680d4f6f146aca0857d4-1664x1057.png?fm=webp"
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Text in nested container Container(Good)- ColumnSet(Default)- Column(attention)- Container(accent+bleed))",
                                            "wrap": true,
                                            "weight": "Bolder",
                                            "size": "Medium",
                                            "spacing": "Medium"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "style": "attention",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://cdn.sanity.io/images/0e4kwcjv/production/0f0a18e6dcf652c09dfd68d5d613499758b9b5a2-1664x1057.png?fm=webp"
                                },
                                {
                                    "type": "TextBlock",
                                    "text": "Another column (Container(Accent)- ColumnSet(Default)- Column(attention)",
                                    "wrap": true,
                                    "size": "Medium",
                                    "weight": "Bolder",
                                    "spacing": "Medium"
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "style": "Accent",
            "bleed": true,
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Container - Style Accent + Bleed",
                    "wrap": true,
                    "weight": "Default",
                    "size": "Medium",
                    "color": "Accent"
                },
                {
                    "type": "ColumnSet",
                    "separator": true,
                    "style": "Good",
                    "bleed": true,
                    "columns": [
                        {
                            "type": "Column",
                            "width": "stretch",
                            "style": "attention",
                            "items": [
                                {
                                    "type": "Container",
                                    "style": "accent",
                                    "bleed": true,
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cdn.sanity.io/images/0e4kwcjv/production/df506530768d939ccccb680d4f6f146aca0857d4-1664x1057.png?fm=webp"
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Text in nested container Container(Accent+Bleed)- ColumnSet(Good+Bleed)- Column(attention)- Container(accent+bleed))",
                                            "wrap": true,
                                            "weight": "Bolder",
                                            "size": "Medium",
                                            "spacing": "Medium"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "style": "attention",
                            "bleed": true,
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://cdn.sanity.io/images/0e4kwcjv/production/0f0a18e6dcf652c09dfd68d5d613499758b9b5a2-1664x1057.png?fm=webp"
                                },
                                {
                                    "type": "TextBlock",
                                    "text": "Another column (Container(Accent+Bleed)- ColumnSet(Good+Bleed)- Column(attention+bleed)",
                                    "wrap": true,
                                    "size": "Medium",
                                    "weight": "Bolder",
                                    "spacing": "Medium"
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "style": "Accent",
            "bleed": true,
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Container - Style Accent + Bleed",
                    "wrap": true,
                    "weight": "Default",
                    "size": "Medium",
                    "color": "Accent"
                },
                {
                    "type": "ColumnSet",
                    "separator": true,
                    "style": "Good",
                    "bleed": true,
                    "columns": [
                        {
                            "type": "Column",
                            "width": "stretch",
                            "style": "attention",
                            "items": [
                                {
                                    "type": "Container",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cdn.sanity.io/images/0e4kwcjv/production/df506530768d939ccccb680d4f6f146aca0857d4-1664x1057.png?fm=webp"
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Text in nested container Container(Accent+Bleed)- ColumnSet(Good+Bleed)- Column(attention)- Container(notset))",
                                            "wrap": true,
                                            "weight": "Bolder",
                                            "size": "Medium",
                                            "spacing": "Medium"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "bleed": true,
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://cdn.sanity.io/images/0e4kwcjv/production/0f0a18e6dcf652c09dfd68d5d613499758b9b5a2-1664x1057.png?fm=webp"
                                },
                                {
                                    "type": "TextBlock",
                                    "text": "Another column (Container(Accent+Bleed)- ColumnSet(Good+Bleed)- Column(NotSet+Bleed)",
                                    "wrap": true,
                                    "size": "Medium",
                                    "weight": "Bolder",
                                    "spacing": "Medium"
                                }
                            ]
                        }
                    ]
                }
            ]
        }
    ],
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.3"
}`

# How Verified
How you verified the fix, including one or all of the following:
1. Manually tested with provided JSON's in both Android and iOS

# Screenshots

### CODE REVIEW

File: source/community/reactnative/src/components/containers/container-wrapper.js [hunk 1]
```diff
@@ -64,13 +64,19 @@ export class ContainerWrapper extends React.PureComponent {
...
+        const padding = this.props.configManager.hostConfig.getEffectiveSpacing(
+            Enums.Spacing.Padding,
+        );
         return (
             <React.Fragment>
                 <BackgroundImage
                     backgroundImage={this.payload.backgroundImage}
                 />
-                <View style={{paddingTop: padding, paddingBottom: padding}}>
+                <View
```
Comment (pavankumarj) [inline source/community/reactnative/src/components/containers/container-wrapper.js:75]: Check if padding needs to be added in all the scenarios(or only backgroundimage)

Comment (pragadeeshk) [reply source/community/reactnative/src/components/containers/container-wrapper.js:75]: @pavankumarj Rest of the changes looks good to me. Please verify backgraoundImage scenario alone. Once confirmed, I will approve the PR

**Review by IbrahimSulai — APPROVED** [review]
LGTM. I will test the target branch once it's merged.

File: source/community/reactnative/src/components/containers/container-wrapper.js (hunk 1 above)
Comment (pavankumarj) [reply source/community/reactnative/src/components/containers/container-wrapper.js:75]: I have added a check for backgroundImage support and that will take care of additional padding and alignment requirements

**Review by pragadeeshk — APPROVED** [review]
LGTM

**Pull request merged**